doc/sphinx/developer_manual/contributing_to_the_code/tips_and_tricks.md

```diff
@@ -4,4 +4,79 @@ Tips and tricks
 
 ```{todo}
 Give some tick and tricks, like start copying plugins which already so almost what you want.
-```
\ No newline at end of file
+```
+
+# Tricks to Run Tests
+
+## Trick 1: Run All Tests
+
+This command would compile and run all the tests.
+
+    make test
+
+Not so much information is given for the failing tests, so the next step is needed for more information.
+
+## Trick 2: Run one test with more information
+
+This command would run one test and print more information to the screen:
+
+    ctest -V -R name_of_test
+
+This is useful to check for failing tests. At the end of the message, the differences between the generated file of the test and the standard file would be shown.
+
+```{todo}
+Give a workflow to generate new tests
+```
+
+## Trick 3: Run the unit_tests separately
+
+In trick 1, the unit_tests are grouped as one big test. Again, not so much information is given about the individual unit tests.
```

Comment:
You can actually list all test cases (`./bin/unit_test --list-test-cases`) and then run an individual test case (`./bin/unit_test --test-case="name of test case"`), or even exclude some test (`./bin/unit_test --test-case-exclude="name of test case"`). For more info use `./bin/unit_test --help`.

Reply:
I have added this information